extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientResponseFilter.java

```diff
@@ -29,21 +33,54 @@ public MicroProfileRestClientResponseFilter(List<ResponseExceptionMapper<?>> exc
             if (exceptionMapper.handles(responseContext.getStatus(), responseContext.getHeaders())) {
-                // we have an exception mapper, we don't need the response anymore, we can map it to response right away (I hope :D)
                 RestClientRequestContext restClientContext = ((ClientRequestContextImpl) requestContext)
                         .getRestClientRequestContext();
-                ResponseImpl response = ClientResponseCompleteRestHandler.mapToResponse(restClientContext, false);
-                Throwable throwable;
-                if (exceptionMapper instanceof ResteasyReactiveResponseExceptionMapper) {
-                    throwable = ((ResteasyReactiveResponseExceptionMapper) exceptionMapper).toThrowable(response,
-                            restClientContext);
+
+                boolean requiresBlocking = RestClientRecorder.isClassBlocking(exceptionMapper.getClass());
+                if (Context.isOnEventLoopThread() && requiresBlocking) {
+                    switchToWorkerThreadPoolAndRetry(restClientContext);
+                    break;
                 } else {
-                    throwable = exceptionMapper.toThrowable(response);
-                }
-                if (throwable != null) {
-                    throw new UnwrappableException(throwable);
+                    // we have an exception mapper, we don't need the response anymore, we can map it to response right away (I hope :D)
+                    ResponseImpl response = ClientResponseCompleteRestHandler.mapToResponse(restClientContext, false);
+                    Throwable throwable;
+                    if (exceptionMapper instanceof ResteasyReactiveResponseExceptionMapper) {
+                        throwable = ((ResteasyReactiveResponseExceptionMapper) exceptionMapper).toThrowable(response,
+                                restClientContext);
+                    } else {
+                        throwable = exceptionMapper.toThrowable(response);
+                    }
+                    if (throwable != null) {
+                        throw new UnwrappableException(throwable);
+                    }
                 }
             }
         }
     }
+
+    private void switchToWorkerThreadPoolAndRetry(RestClientRequestContext restClientContext) {
+        int position = restClientContext.getPosition();
+
+        List<ClientRestHandler> nextHandlers = new ArrayList<>(3 + restClientContext.getHandlers().length - position);
+        nextHandlers.add(new ClientUseWorkerExecutorRestHandler());
+        nextHandlers.add(currentHandler(restClientContext));
+        nextHandlers.add(new ClientResetExecutorRestHandler());
+
+        while (position < restClientContext.getHandlers().length) {
+            nextHandlers.add(restClientContext.getHandlers()[position]);
+            position++;
+        }
+
+        restClientContext.restart(nextHandlers.toArray(new ClientRestHandler[0]), true);
```

Comment:
Last minor content: `new ClientRestHandler[0]` can be a constant :)

Reply:
Done